tests/utils/managed_utils/mod.rs

```diff
@@ -0,0 +1,583 @@
+#![allow(dead_code)]
+/// We allow dead code here due to the following issue:
+/// https://github.com/rust-lang/rust/issues/46379, which would otherwise create
+/// a lot of unused code warnings.
+use openmls::prelude::{node::Node, *};
+use rand::{rngs::OsRng, RngCore};
+use std::{cell::RefCell, collections::HashMap};
+
+pub mod client;
+pub mod default_callbacks;
+pub mod errors;
+
+use self::client::*;
+use self::errors::*;
+
+#[derive(Debug)]
+pub struct KeyStore {
+    // Maps a client Id and a ciphersuite to a CredentialBundle.
+    credential_bundles: HashMap<(Vec<u8>, CiphersuiteName), CredentialBundle>,
+}
+
+impl<'ks> KeyStore {
+    pub(crate) fn store_credentials(
+        &mut self,
+        client_id: Vec<u8>,
+        credential_bundles: Vec<(CiphersuiteName, CredentialBundle)>,
+    ) {
+        for (cn, cb) in credential_bundles {
+            self.credential_bundles.insert((client_id.clone(), cn), cb);
+        }
+    }
+
+    pub(crate) fn get_credential(
+        &self,
+        client_id: &Vec<u8>,
+        ciphersuite_name: CiphersuiteName,
+    ) -> Option<&CredentialBundle> {
+        let key = &(client_id.clone(), ciphersuite_name);
+        self.credential_bundles.get(key)
+    }
+}
+
+#[derive(Clone)]
+pub struct Group {
+    pub group_id: GroupId,
+    pub members: Vec<(usize, Vec<u8>)>,
+    pub ciphersuite: Ciphersuite,
+    pub group_config: ManagedGroupConfig,
+    pub public_tree: Vec<Option<Node>>,
+}
+
+impl Group {
+    pub fn random_group_member(&self) -> Vec<u8> {
+        let index = (OsRng.next_u32() as usize) % self.members.len();
+        // We can unwrap here, because the index is scoped with the size of the
+        // HashSet.
+        let (_, identity) = self.members.iter().nth(index).unwrap().clone();
+        identity
+    }
+}
+
+pub enum ActionType {
+    Commit,
+    Proposal,
+}
+
+pub enum OperationType {
+    Update,
+    Add,
+    Remove,
+}
+
+pub struct ManagedTestSetup<'client_lifetime> {
+    pub number_of_clients: usize,
+    // The clients identity is its position in the vector in be_bytes.
```

Comment:
😬 I don't think I like this approach for identities. But it's as good or bad as any other thing we can do here.
We shoudl have a function to get the identity for a client though, i.e. don't duplicate `i.to_be_bytes().to_vec()`.

Reply:
Feel free to suggest another approach. I'm not particularly attached to this one, so I'm happy to replace it with something else. This just seemed convenient to implement.